src/nlp.jl

```diff
@@ -463,9 +771,64 @@ function quadToExpr(q::QuadExpr,constant::Bool)
     return ex
 end
 
+# we splat in the subexpressions (for now)
+function tapeToExpr(k, nd::Vector{NodeData}, adj, const_values, parameter_values, subexpressions::Vector{Any})
+
+    children_arr = rowvals(adj)
+
+    nod = nd[k]
+    if nod.nodetype == VARIABLE
+        return Expr(:ref,:x,nod.index)
+    elseif nod.nodetype == VALUE
+        return const_values[nod.index]
+    elseif nod.nodetype == SUBEXPRESSION
+        return subexpressions[nod.index]
+    elseif nod.nodetype == PARAMETER
+        return parameter_values[nod.index]
+    elseif nod.nodetype == CALL
+        op = nod.index
+        opsymbol = operators[op]
+        children_idx = nzrange(adj,k)
+        ex = Expr(:call,opsymbol)
+        for cidx in children_idx
+            push!(ex.args, tapeToExpr(children_arr[cidx], nd, adj, const_values, parameter_values, subexpressions))
+        end
+        return ex
+    elseif nod.nodetype == CALLUNIVAR
+        op = nod.index
+        opsymbol = univariate_operators[op]
+        cidx = first(nzrange(adj,k))
+        return Expr(:call,opsymbol,tapeToExpr(children_arr[cidx], nd, adj, const_values, parameter_values, subexpressions))
+    elseif nod.nodetype == COMPARISON
+        op = nod.index
+        opsymbol = comparison_operators[op]
+        children_idx = nzrange(adj,k)
+        ex = Expr(:comparison)
+        for cidx in children_idx
+            push!(ex.args, tapeToExpr(children_arr[cidx], nd, adj, const_values, parameter_values, subexpressions))
+            push!(ex.args, opsymbol)
+        end
+        pop!(ex.args)
+        return ex
+    elseif nod.nodetype == LOGIC
+        op = nod.index
+        opsymbol = logic_operators[op]
+        children_idx = nzrange(adj,k)
+        lhs = tapeToExpr(children_arr[first(children_idx)], nd, adj, const_values, parameter_values, subexpressions)
+        rhs = tapeToExpr(children_arr[last(children_idx)], nd, adj, const_values, parameter_values, subexpressions)
+        return Expr(opsymbol, lhs, rhs)
+    else
+        error()
```

Comment:
In general, rather than this:

```julia
else
    error()
end
```

it's preferred to do this
```julia
# skip else
end
error()
```

This is because it still triggers an error if one of the preceding statements forgot to include a `return`.